src/App.tsx

```diff
@@ -16,35 +16,101 @@
 ];
 
 export const App: React.FC = () => {
+  const [sort, setSort] = useState<string[]>([...goodsFromServer]);
+  const [reverse, setReverse] = useState<boolean>(false);
+  const [active, setActive] = useState<string>('');
+
+  const handleSort = (good: string): void => {
+    let newSort: string[];
+
+    switch (good) {
+      case 'abc':
+        if (reverse) {
+          newSort = [...sort].sort((a, b) => b.localeCompare(a));
+          setActive(good);
+        } else {
+          newSort = [...sort].sort((a, b) => a.localeCompare(b));
+          setActive(good);
+        }
+
+        break;
+      case 'length':
+        if (reverse) {
+          newSort = [...sort].sort((a, b) => b.length - a.length);
+          setActive(good);
+        } else {
+          newSort = [...sort].sort((a, b) => a.length - b.length);
+          setActive(good);
+        }
+
+        break;
+      case 'reverse':
+        if (active === 'abc' || active === 'length') {
+          setReverse(!reverse);
+        } else if (active === 'reverse') {
+          setReverse(!reverse);
+          setActive('');
+        } else {
+          setReverse(!reverse);
+          setActive('reverse');
+        }
+
+        newSort = [...sort].reverse();
+        break;
+      default:
+        setActive('');
+        setReverse(false);
+        newSort = [...goodsFromServer];
+    }
+
+    setSort(newSort);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${(active !== 'abc' ? 'is-light' : '') || (active !== 'abc' && reverse) ? 'is-light' : ''}`}
+          onClick={() => handleSort('abc')}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={`button is-success ${(active !== 'length' ? 'is-light' : '') || (active !== 'length' && reverse) ? 'is-light' : ''}`}
```

Comment:
The className logic for the 'Sort by length' button is also unnecessarily complex and might not work as intended. Similar to the previous comment, the condition `(active !== 'length' ? 'is-light' : '') || (active !== 'length' && reverse) ? 'is-light' : ''` will always evaluate to 'is-light' when `active` is not 'length'. Simplify this logic to ensure the correct class is applied.